### PR TITLE
[COS-838] - COS API - Add service to contract

### PR DESCRIPTION
**Pull request by grokkos**
## Changes

What types of changes does your code introduce?  _Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no API changes)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Build related changes
- [ ] Documentation Update (if none of the other choices apply)
- [ ] Other (please describe below)

## Summary by CodeRabbit

- **New Features**
  - Introduced a new search and creation functionality for service line items within the application.
  - Added a new `BilledType` enum to classify service line items and associated GraphQL schema updates.
  - Implemented new GraphQL queries and mutations for service line items.

- **Enhancements**
  - Added new data loader for efficient batch loading of service line items.
  - Expanded the GraphQL schema with new types and enums for service line items.

- **Testing**
  - Created new integration test placeholders for service line item creation.

- **Internal Improvements**
  - Developed new mappers for converting between entity and model representations of service line items.
  - Established new repository interfaces and implementations for service line item data interactions.
  - Integrated OpenTracing for enhanced observability within service line item operations.

- **Documentation**
  - Updated GraphQL schema documentation to reflect new service line item types and operations.

### CODE REVIEW

File: packages/server/customer-os-api/entity/billed_type.go [hunk 1]
```diff
@@ -0,0 +1,33 @@
+package entity
+
+type BilledType string
+
+const (
+	BilledTypeNone     BilledType = ""
+	BilledTypeMonthly  BilledType = "MONTHLY_BILLED"
```
Comment (alexopenline) [inline packages/server/customer-os-api/entity/billed_type.go]: In DB I have these values:
"MONTHLY", "ANNUALLY", "ONCE"

File: packages/server/customer-os-api/entity/service_line_item.go [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+import "time"
+
+type ServiceLineItemEntity struct {
+	ID            string
+	Name          string
+	CreatedAt     time.Time
+	UpdatedAt     time.Time
+	Billed        BilledType
+	Price         float64
+	Quantity      int
```
Comment (alexopenline) [inline packages/server/customer-os-api/entity/service_line_item.go]: in DB it's int64

File: packages/server/customer-os-api/repository/service_line_item_repository.go [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+	})
+	return utils.NodePtr(dbRecord.(*db.Record).Values[0].(dbtype.Node)), err
+}
+
+func (r *serviceLineItemRepository) GetForContracts(ctx context.Context, tenant string, contractIds []string) ([]*utils.DbNodeAndId, error) {
+	span, ctx := opentracing.StartSpanFromContext(ctx, "ServiceLineItemRepository.GetForContracts")
+	defer span.Finish()
+	tracing.SetDefaultNeo4jRepositorySpanTags(ctx, span)
+
+	cypher := `MATCH (t:Tenant {name:$tenant})<-[:CONTRACT_BELONGS_TO_TENANT]-(c:Contract)-[:HAS_SERVICE_LINE_ITEM]->(serviceLineItem:ServiceLineItem)-[:SERVICE_LINE_ITEM_BELONGS_TO_TENANT]->(t)
```
Comment (alexopenline) [inline packages/server/customer-os-api/repository/service_line_item_repository.go]: This relation between service line item and tenant is not available

Comment (alexopenline) [inline packages/server/customer-os-api/repository/service_line_item_repository.go]: HAS_SERVICE_LINE_ITEM should be HAS_SERVICE

See ServiceLineItemRepository.CreateForContract for exact query creation

File: packages/server/customer-os-api/service/service_line_item_service.go [hunk 4]
```diff
@@ -0,0 +1,167 @@
...
+
+	switch serviceLineItemDetails.ServiceLineItemEntity.Billed {
+	case entity.BilledTypeMonthly:
+		createServiceLineItemRequest.Billed = servicelineitempb.BilledType_MONTHLY_BILLED
+	case entity.BilledTypeAnnually:
+		createServiceLineItemRequest.Billed = servicelineitempb.BilledType_ANNUALLY_BILLED
+	case entity.BilledTypeOnce:
+		createServiceLineItemRequest.Billed = servicelineitempb.BilledType_ONCE_BILLED
+	default:
+		//TODO add to service_line_item proto billed type NONE_BILLED and use it as default here
```
Comment (alexopenline) [inline packages/server/customer-os-api/service/service_line_item_service.go]: there is no none option so far, there should be an option mandatory sent on create

File: packages/server/customer-os-api/entity/billed_type.go (hunk 1 above)
Comment (grokkos) [reply packages/server/customer-os-api/entity/billed_type.go]: okay fixing :)

File: packages/server/customer-os-api/repository/service_line_item_repository.go (hunk 3 above)
Comment (grokkos) [reply packages/server/customer-os-api/repository/service_line_item_repository.go]: thanks

**Review by alexopenline — APPROVED** [review]

**Pull request merged**